src/main/java/community/solace/spring/boot/starter/solaceclientconfig/KeyStoreFactory.java

```diff
@@ -34,14 +34,31 @@ KeyStore createClientKeyStore(final String privateKeyPem, final String clientCer
             if (certificates == null) {
                 return null;
             }
+
+            if (!privateKeyMatchesCertificate(privateKey, certificates)) {
+                LOG.error("Non of the given certificates in SSL_CLIENT_CERT matches the give key SSL_PRIVATE_KEY");
+                return null;
+            }
+
             keyStore.setKeyEntry("pk", privateKey, INTERNAL_PASSWORD.toCharArray(), certificates);
+
             return keyStore;
         } catch (Exception exception) {
             LOG.warn("Error during keystore creation", exception);
             return null;
         }
     }
 
+    private boolean privateKeyMatchesCertificate(PrivateKey privateKey, Certificate[] certificates) {
+        for (Certificate certificate : certificates) {
```

Comment:
Only the first certificate is the user's certificate where the private key must match. Additional certificates are intermediates where the private key is not used. See [JavaDoc of KeyStore](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/security/KeyStore.html#getCertificateChain(java.lang.String)):

> the certificate chain (ordered with the user's certificate first followed by zero or more certificate authorities)